tests/test_certgen.py

```diff
@@ -0,0 +1,168 @@
+import OpenSSL
+import boto3
+import botocore.session
+import configargparse
+import jks
+import logging
+import pytest
+from acm_pca_cert_generator import certgen
+from botocore.stub import Stubber
+from collections import Counter
+
+
+def test_check_key_length_valid():
+    assert certgen.check_key_length("2048") == 2048
+
+
+def test_check_key_length_not_number():
+    with pytest.raises(configargparse.ArgumentTypeError):
+        certgen.check_key_length("test")
+
+
+def test_check_key_length_invalid_choice():
+    with pytest.raises(configargparse.ArgumentTypeError):
+        certgen.check_key_length("1024")
+
+
+def test_check_validity_period_valid():
+    assert certgen.check_validity_period("365d") == "365d"
+    assert certgen.check_validity_period("12m") == "12m"
+    assert certgen.check_validity_period("1y") == "1y"
+
+
+def test_check_validity_period_invalid():
+    with pytest.raises(configargparse.ArgumentTypeError):
+        certgen.check_validity_period("52w")
+
+
+def test_generate_private_key():
+    pkey = certgen.generate_private_key("RSA", 2048)
+    assert pkey.type() == OpenSSL.crypto.TYPE_RSA
+    assert pkey.bits() == 2048
+
+
+def test_generate_private_key_invalid_type():
+    with pytest.raises(ValueError):
+        pkey = certgen.generate_private_key("TSA", 2048)
+
+
+def test_generate_private_key_invalid_length():
+    with pytest.raises(TypeError):
+        pkey = certgen.generate_private_key("RSA", "notanint")
```

Comment:
Is this different to test_check_key_length_not_number()? If the user typed a notanint it should have been handled by the arg parser. Also, the code generate_private_key method is not explicitly doing type checking and it is the library you're calling doing it. Should your method maybe do the checking (if you're going to test it)?

Reply:
Oh, and you can remove the unused variable.